core/modules/installer/installer.browser.inc

```diff
@@ -0,0 +1,470 @@
+      return array();
+    }
+  }
+
+  // Set the cached version of the results.
+  cache_set($cid, $results, 'cache', strtotime("+24 hours"));
+
+  return $results;
+}
+
+/**
+ * Gets the server to use for fetching results.
+ *
+ * @return
+ *   Returns the active server, in the format:
+ *   array(
+ *     "url" => "https://projects.backdropcms.org",
+ *     "name" => "Backdrop",
+ *     "method" => "json"
+ *   )
+ */
+function installer_browser_get_server() {
+  $server = config_get('installer.settings', 'installer_browser_server');
+  backdrop_alter('installer_browser_server', $server);
+
+  return $server;
+}
+
+/**
+ * Uses the project status url to get the available releases for a project.
+ *
+ * @param $project
+ *   The project to get the releases for.
+ *
+ * @return
+ *   An array of releases for this project.
+ */
+function installer_browser_get_project_release_data($project) {
+  $project['project_type'] = $project['type'];
+  // Build the releases cache for this project
+  module_load_include('inc', 'update', 'update.fetch');
+  if (_update_process_fetch_task($project)) {
+    $data = _update_cache_get('available_releases::' . $project['name']);
+    if (isset($data->data) AND isset($data->data['releases']) AND is_array($data->data['releases'])) {
+      return $data->data;
+    }
+  }
+
+  return FALSE;
+}
+
+/**
+ * Installs a single release of a project during batch.
+ */
+function _installer_browser_batch_install_release($release_name, $project, &$context) {
+  $release = installer_browser_get_release($release_name, $project);
+
+  module_load_include('inc', 'installer', 'installer.manager');
+  $url = file_create_url($release['download_link']);
+  $result = installer_manager_download_project($url, TRUE);
+
+  if ($result['success']) {
+    $context['results']['successes'][] = t('Successfully installed %project.', array('%project' => $project['title']));
+    $context['message'] = t('Installed %project...', array('%project' => $project['title']));
+
+    // Add this to the session variable and remove it from the install_queue
+    // variable.
+    $_SESSION['installer_browser_installed_projects'][$project['name']] = $project;
+    unset($_SESSION['installer_browser_install_list'][$project['name']]);
+  }
+  else {
+    watchdog('project_browser', 'There was an error while installing %project.
+      !message',
+      array('%project' => $project['title'], '!message' => $result['message']), WATCHDOG_ERROR);
+    $context['results']['failures'][] = t('Error installing %project. Errors have been logged.',
+      array('%project' => $project['title']));
+    $context['message'] = t('Error installing %project. !message',
+      array('%project' => $project['title'], '!message' => $result['message']));
+  }
+}
+
+/**
+ * Shows a message and finish up the batch.
+ */
+function _installer_browser_batch_install_releases_finished($success, $results, $operations) {
+  backdrop_get_messages();
```

Comment:
Hm, intentionally wipe out any messages? Could use a code comment if this is intentional.